isthmus/src/main/java/io/substrait/isthmus/expression/RexExpressionConverter.java

```diff
@@ -73,7 +92,115 @@ public Expression visitLiteral(RexLiteral literal) {
+    var upperBound = toWindowBound(over.getWindow().getUpperBound());
+    var sqlAggFunction = over.getAggOperator();
+    var argList = over.getOperands().stream().map(r -> ((RexSlot) r).getIndex()).toList();
+    boolean approximate = false;
+    int filterArg = -1;
+    var aggCall =
+        AggregateCall.create(
+            sqlAggFunction,
+            over.isDistinct(),
+            approximate,
+            over.ignoreNulls(),
+            argList,
+            filterArg,
+            null,
+            RelCollations.EMPTY,
+            over.getType(),
+            sqlAggFunction.getName());
+    var builder = Expression.Window.builder();
+    AbstractFunctionInvocation functionInvocation = null;
+    for (var converter : functionConverters) {
+      var funcInvocationOpt =
+          converter.convert(
+              inputRel,
+              inputType,
+              aggCall,
+              t -> {
+                RexNode r = (RexNode) t;
+                return r.accept(this);
+              });
+      if (funcInvocationOpt.isPresent()) {
+        functionInvocation = (AbstractFunctionInvocation) funcInvocationOpt.get();
+        if (functionInvocation instanceof AggregateFunctionInvocation) {
+          var aggFuncInvocation = (AggregateFunctionInvocation) functionInvocation;
+          var aggMeasure = Aggregate.Measure.builder().function(aggFuncInvocation).build();
+          builder.aggregateFunction(aggMeasure).hasNormalAggregateFunction(true);
+        } else if (functionInvocation instanceof WindowFunctionInvocation) {
+          var windowFuncInvocation = (WindowFunctionInvocation) functionInvocation;
+          var windowFunc =
+              ImmutableExpression.WindowFunction.builder().function(windowFuncInvocation).build();
+          builder.windowFunction(windowFunc).hasNormalAggregateFunction(false);
+        } else {
+          throw new RuntimeException(
+              String.format("Unexpected AbstractFunctionInvocation :%s", functionInvocation));
+        }
+      }
+    }
+    if (functionInvocation == null) {
+      throw new RuntimeException(
+          String.format(
+              "Not found the corresponding window aggregate function:%s", sqlAggFunction));
+    }
+    var window = over.getWindow();
+    var partitionExps = window.partitionKeys.stream().map(r -> r.accept(this)).toList();
+    var sortFields = window.orderKeys.stream().map(r -> toSortField(r)).toList();
+
+    return builder
+        .addAllOrderBy(sortFields)
+        .addAllPartitionBy(partitionExps)
+        .lowerBound(lowerBound)
+        .upperBound(upperBound)
+        .type(TypeConverter.convert(over.getType()))
+        .build();
+  }
+
+  private WindowBound toWindowBound(RexWindowBound rexWindowBound) {
+    if (rexWindowBound.isCurrentRow()) {
+      return ImmutableWindowBound.CurrentRowWindowBound.builder().build();
+    }
+    if (rexWindowBound.isUnbounded()) {
+      var direction =
```

Comment:
line 166 - 169 code seems to be same as line 173 - 175. Can we refactor to remove the duplicate?